### PR TITLE
[Release] Update Release Process Documentation

### PR DESCRIPTION
**Pull request by mfitton**
## Why are these changes needed?
Updates the release process documentation with learnings from the latest release process and info about the ray-project/releaser project

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by mfitton** [comment]
@wuisawesome @simon-mo your reviews are optional, but much appreciated!

Please let me know if you think there's anything I've missed.

**Comment by mfitton** [comment]
There were initially some formatting issues, hopefully I've resolved them all now.

**Review by simon-mo — APPROVED** [review]
LGTM

File: release/RELEASE_PROCESS.rst [hunk 1]
```diff
@@ -1,45 +1,127 @@
...
+Before Branch Cut
+-----------------
+1. **Create a document to track release-blocking commits.** These may be pull
+   requests that are not ready at the time of branch cut, or they may be
+   fixes for issues that you encounter during release testing later.
+   The only PRs that should be considered release-blocking are those which
+   fix a MAJOR REGRESSION (P0) or deliver an absolutely critical piece of
+   functionality that has been promised for the release (though this should
+   be avoided where possible).
+   You may make a copy of the following `template <https://docs.google.com/spreadsheets/d/1qeOYErAn3BzGgtEilBePjN6tavdbabCEEqglDsjrq1g/edit#gid=0>`_.
```
Comment (edoakes) [inline release/RELEASE_PROCESS.rst:19]: Make sure that this is a publicly-accessible document with no internal comments on it

File: release/RELEASE_PROCESS.rst [hunk 2]
```diff
@@ -1,45 +1,127 @@
...
+
+   Make sure to share this document with the folks at Ant Financial
+   in our open-source collaboration channel.
```
Comment (edoakes) [inline release/RELEASE_PROCESS.rst:22]:
```suggestion

   Make sure to share this document with major contributors who may have release blockers.
```

**Review by edoakes — APPROVED** [review]
Great improvements, thanks for doing this.

**Review by wuisawesome — APPROVED** [review]

**Pull request merged**